app/Actions/Album/Create.php

```diff
@@ -85,4 +105,35 @@ private function copyPermission(?Album $parentAlbum): array
 
 		return $copyPermissions;
 	}
+
+	/**
+	 * If album is created by someone who has the album shared with.
+	 * We need to give access all to that person.
+	 *
+	 * @param Album $album
+	 *
+	 * @return void
+	 */
+	private function fixPermissionIfNotOwner(Album $album)
+	{
+		if ($album->owner_id === $this->intendedOwnerId) {
+			return;
+		}
+
+		$album->access_permissions()
+			->where(APC::USER_ID, '=', $this->intendedOwnerId)
+			->where(APC::BASE_ALBUM_ID, '=', $album->id)
+			->delete();
+
+		$accessPerm = new AccessPermission([
```

Comment:
Maybe we could add a static constructor in the AccessPermission class to have with only an id as parameter and which put all grants at true ?

Reply:
Fixed : [d4e2bdf](https://github.com/LycheeOrg/Lychee/pull/2298/commits/d4e2bdf48204ecc6264b7a88cdfdd85cc26e536b)